src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,131 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+
+  public abstract ImmutableList<String> comments();
+
+  public abstract ImmutableList<String> params();
+
+  public abstract String deprecated();
```

Comment:
Note that it's optional whether we have `deprecated()` and `throwType()` getters.

Reply:
I think we need these getters because `comment()` calls them to get `@deprecated` and `@throws`, please correct me if wrong.

Reply:
Instead, we can have "private" (in the sense we've been using in these AutoBuilders) getters in the Builder class.

Reply:
got it, but even if we use `private` getters in the Builder class, we still need a `private` getter as a class field. For example `name` filed in MethodDefinition [class](https://github.com/googleapis/gapic-generator-java/blob/master/src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java#L60) and private [getter](https://github.com/googleapis/gapic-generator-java/blob/master/src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java#L128).

Reply:
Have you tried something like the following? I would recommend doing more investigation and trying more approaches next time.

`public abstract static class Builder {
    String throwType = null;
    String throwDescription = null;
    
    public Builder setThrows(String type, String description) {
      throwType = type;
      throwDescription = description;
    }
    // Similar for setDeprecated and params.
    
    public JavaDocComment build() {
      // Add all other contents.
      // Add params, throws, and deprecated.
    }
}

Reply:
thanks for the recommendation, I did think about this way, but it would "break" our getter/setter pattern. For learning purpose, as the [name](https://github.com/googleapis/gapic-generator-java/blob/master/src/main/java/com/google/api/generator/engine/ast/MethodDefinition.java#L60) field in MethodDefinition class, why we use private getter there, but local definition here.